File: tools/generate_icon.py
from PIL import Image, ImageDraw, ImageFont
import os

def create_wallet_icon(size=1024):
    # Criar imagem com fundo branco
    image = Image.new('RGB', (size, size), 'white')
    draw = ImageDraw.Draw(image)
    
    # Desenhar carteira (retângulo vermelho arredondado)
    margin = size * 0.15
    wallet_width = size * 0.7
    wallet_height = size * 0.5
    wallet_x = margin
    wallet_y = size * 0.25
    
    draw.rounded_rectangle(
        [(wallet_x, wallet_y), 
         (wallet_x + wallet_width, wallet_y + wallet_height)],
        radius=size * 0.1,
        fill='#EF5350'  # Material Red 400
    )
    
    # Desenhar notas de dinheiro (retângulos brancos)
    note_width = size * 0.5
    note_height = size * 0.1
    
    draw.rounded_rectangle(
        [(size * 0.25, size * 0.2),
         (size * 0.25 + note_width, size * 0.2 + note_height)],
        radius=size * 0.02,
        fill='white'
    )
    
    draw.rounded_rectangle(
        [(size * 0.2, size * 0.15),
         (size * 0.2 + note_width, size * 0.15 + note_height)],
        radius=size * 0.02,
        fill='white'
    )
    
    # Tentar carregar uma fonte
    try:
        font = ImageFont.truetype("Arial", size=int(size * 0.2))
    except:
        font = ImageFont.load_default()
    
    # Desenhar símbolo R$
    text = "R$"
    text_width = draw.textlength(text, font=font)
    text_x = (size - text_width) / 2
    text_y = size * 0.35
    
    draw.text(
        (text_x, text_y),
        text,
        font=font,
        fill='white'
    )
    
    # Salvar a imagem
    os.makedirs('assets/icon', exist_ok=True)
    image.save('assets/icon/icon.png', 'PNG')
    
    # Criar tamanhos para Android
    sizes = {
        'mdpi': 48,
        'hdpi': 72,
        'xhdpi': 96,
        'xxhdpi': 144,
        'xxxhdpi': 192
    }
    
    for dpi, dpi_size in sizes.items():
        resized = image.resize((dpi_size, dpi_size), Image.Resampling.LANCZOS)
        os.makedirs(f'android/app/src/main/res/mipmap-{dpi}', exist_ok=True)
        resized.save(f'android/app/src/main/res/mipmap-{dpi}/ic_launcher.png', 'PNG')

if __name__ == '__main__':
    create_wallet_icon()

File: bin/generate_icon.dart
import 'dart:io';
import 'dart:math';
import 'package:image/image.dart' as img;

void main() async {
  await generateIcon();
}

Future<void> generateIcon() async {
  // Criar uma imagem 1024x1024
  final image = img.Image(width: 1024, height: 1024);

  // Preencher com fundo branco
  img.fillRect(image,
      x1: 0,
      y1: 0,
      x2: image.width - 1,
      y2: image.height - 1,
      color: img.ColorInt32.rgb(255, 255, 255));

  // Desenhar carteira (retângulo roxo)
  final walletColor = img.ColorInt32.rgb(103, 80, 164); // #6750A4
  final walletX = (image.width * 0.15).round();
  final walletY = (image.height * 0.25).round();
  final walletWidth = (image.width * 0.7).round();
  final walletHeight = (image.height * 0.5).round();

  drawRoundedRect(image,
      x: walletX,
      y: walletY,
      width: walletWidth,
      height: walletHeight,
      radius: (image.width * 0.1).round(),
      color: walletColor);

  // Desenhar notas de dinheiro (retângulos brancos)
  final whiteColor = img.ColorInt32.rgb(255, 255, 255);
  final note1X = (image.width * 0.25).round();
  final note1Y = (image.height * 0.2).round();
  final noteWidth = (image.width * 0.5).round();
  final noteHeight = (image.height * 0.1).round();

  drawRoundedRect(image,
      x: note1X,
      y: note1Y,
      width: noteWidth,
      height: noteHeight,
      radius: (image.width * 0.02).round(),
      color: whiteColor);

  final note2X = (image.width * 0.2).round();
  final note2Y = (image.height * 0.15).round();

  drawRoundedRect(image,
      x: note2X,
      y: note2Y,
      width: noteWidth,
      height: noteHeight,
      radius: (image.width * 0.02).round(),
      color: whiteColor);

  // Criar diretórios se não existirem
  final iconDir = Directory('android/app/src/main/res/mipmap-xxxhdpi');
  if (!await iconDir.exists()) {
    await iconDir.create(recursive: true);
  }

  // Salvar a imagem
  final iconBytes = img.encodePng(image);
  final iconFile = File('${iconDir.path}/ic_launcher.png');
  await iconFile.writeAsBytes(iconBytes);

  print('Ícone gerado com sucesso em: ${iconFile.path}');
}

void drawRoundedRect(img.Image image,
    {required int x,
    required int y,
    required int width,
    required int height,
    required int radius,
    required img.ColorInt32 color}) {
  // Desenhar retângulo com cantos arredondados
  for (int i = x; i < x + width; i++) {
    for (int j = y; j < y + height; j++) {
      // Verificar se o pixel está dentro do retângulo arredondado
      if (isInsideRoundedRect(i - x, j - y, width, height, radius)) {
        img.drawPixel(image, i, j, color);
      }
    }
  }
}

bool isInsideRoundedRect(int x, int y, int width, int height, int radius) {
  // Verificar se o ponto está dentro do retângulo com cantos arredondados
  if (x < radius) {
    if (y < radius) {
      return sqrt(pow(x - radius, 2) + pow(y - radius, 2)) <= radius;
    } else if (y > height - radius) {
      return sqrt(pow(x - radius, 2) + pow(y - (height - radius), 2)) <= radius;
    }
  } else if (x > width - radius) {
    if (y < radius) {
      return sqrt(pow(x - (width - radius), 2) + pow(y - radius, 2)) <= radius;
    } else if (y > height - radius) {
      return sqrt(
              pow(x - (width - radius), 2) + pow(y - (height - radius), 2)) <=
          radius;
    }
  }
  return x >= 0 && x < width && y >= 0 && y < height;
}
